### PR TITLE
Add api for analytical_statement from analysis

### PR DESCRIPTION
**Pull request by k9845**
fixes:
- #662

### CODE REVIEW

File: apps/analysis/tests/test_apis.py [hunk 1]
```diff
@@ -89,6 +89,62 @@ def test_create_pillar_from_analysis(self):
...
+                    'statement': "tea",
+                    'order': 2
+                }
+            ]
+        }
+        self.authenticate(user)
+        url = f'/api/v1/projects/{project.id}/analysis/{analysis.id}/pillars/{response_id}/'
+        response = self.client.patch(url, data)
+        self.assert_200(response)
+        self.assertEqual(response.data['main_statement'], data['main_statement'])
```
Comment (annshress) [inline apps/analysis/tests/test_apis.py:168]: Add test:

Check if old analytical statements is/should be retained.
Check new analytical statements count.

Comment (k9845) [reply apps/analysis/tests/test_apis.py:168]: Done, thanks !!

File: apps/analysis/serializers.py [hunk 2]
```diff
@@ -16,9 +16,38 @@
...
+class AnalyticalStatementSerializer(RemoveNullFieldsMixin,
+                                    DynamicFieldsMixin,
+                                    NestedCreateMixin,
+                                    NestedUpdateMixin,):
+    analytical_entries = AnlyticalEntriesSerializer(source='analyticalstatemententry_set', many=True, required=False)
+
+    class Meta:
+        model = AnalyticalStatement
+        fields = '__all__'
+        read_only_fields = ('analysis_pillar', )
```
Comment (thenav56) [inline apps/analysis/serializers.py]:
```suggestion
        read_only_fields = ('analysis_pillar',)
```

File: apps/analysis/views.py [hunk 3]
```diff
@@ -112,6 +112,34 @@ def get_queryset(self):
...
+                context={'request': request, 'view': self},
+                partial=True,
+            )
+            serializer.is_valid()
+            serializer.update(statement, statement_maps[statement.id])
+
+        return response.Response(
+            AnalysisPillarSerializer(instance, context={'request': request, 'view': self}).data,
+        )
+
```
Comment (thenav56) [inline apps/analysis/views.py]: Let's use Piller API with patch for this.

File: apps/analysis/tests/test_apis.py [hunk 4]
```diff
@@ -201,6 +290,35 @@ def test_clone_analysis(self):
...
+        url = f'/api/v1/projects/{project.id}/analysis/{analysis.id}/pillars/{pillar.id}/statement/'
+        data = {
+            'analytical_statement': [
+                {
+                    'id': analytical_statement.id,
+                    'statement': 'Hello from there'
+                }
+            ]
+        }
+        self.authenticate(user)
```
Comment (thenav56) [inline apps/analysis/tests/test_apis.py]: Let's use pillar API here as well.

Comment (k9845) [reply apps/analysis/tests/test_apis.py]: Done

File: apps/analysis/views.py (hunk 3 above)
Comment (k9845) [reply apps/analysis/views.py]: Done

File: apps/analysis/serializers.py (hunk 2 above)
Comment (k9845) [reply apps/analysis/serializers.py]: Done

**Review by annshress — APPROVED** [review]

File: apps/analysis/tests/test_apis.py [hunk 5]
```diff
@@ -201,6 +290,28 @@ def test_clone_analysis(self):
...
+        analysis = self.create(Analysis)
+        pillar = self.create(AnalysisPillar, analysis=analysis, title='title1', assignee=user)
+        analytical_statement = self.create(AnalyticalStatement, analysis_pillar=pillar, statement='Hello from here')
+
+        url = f'/api/v1/projects/{project.id}/analysis/{analysis.id}/pillars/{pillar.id}/'
+        data = {
+            'analytical_statement': [
+                {
+                    'id': analytical_statement.id,
+                    'statement': 'Hello from there'
```
Comment (thenav56) [inline apps/analysis/tests/test_apis.py]: Maybe add entries here as well?

File: apps/analysis/serializers.py [hunk 6]
```diff
@@ -16,9 +16,38 @@
...
+    class Meta:
+        model = AnalyticalStatement
+        fields = '__all__'
+        read_only_fields = ('analysis_pillar',)
+
+    def validate(self, data):
+        analysis_pillar_id = self.context['view'].kwargs.get('analysis_pillar_id', None)
+        if analysis_pillar_id:
+            data['analysis_pillar_id'] = int(analysis_pillar_id)
+        return data
```
Comment (thenav56) [inline apps/analysis/serializers.py]: Let's make this compulsary like before.

File: apps/analysis/tests/test_apis.py (hunk 5 above)
Comment (k9845) [reply apps/analysis/tests/test_apis.py]: yes

**Review by thenav56 — APPROVED** [review]

**Review by annshress — APPROVED** [review]

**Pull request merged**